BUILD.md

```diff
@@ -0,0 +1,170 @@
+# Build FreeOrion
+
+
+## Hardware and OS Requirements
+
+FreeOrion will compile on most recent Windows, Mac OS X and Linux operating
+systems. Other operating systems have reported to work, but support is not
+actively maintained by the FreeOrion developers. FreeOrion is developed for x86
+compatible processor architectures, other architectures haven't been tested.
+
+The FreeOrion source code makes heavy use of templates and requires much memory
+to compile ; 4GB RAM or more is recommended. To build FreeOrion, expect up to
+45 minutes on a Intel Core i5 system.
+
+
+## Required Software Dependencies
+
+FreeOrion depends on the following software to build:
+
+  * [Visual Studio] - 2015 for Windows Desktop Update 3 ; Windows only
+  * [Xcode] - 6.4 or later ; Mac OS X only
+  * [CMake] - 3.4 (Mac OS X) ; 3.1 or later (Other non-Windows)
+  * A C++11 compliant compiler - Other Operating Systems
+    * [GNU GCC] - 5.0 or later
+    * [Clang] - 3.9 or later
+  * [Python] - 2 or later, but not 3 or later
+  * [Git]
+
+FreeOrion depends on the following libraries or APIs to run:
+
+  * OpenGL - 2.1 or later ; usually provided by the graphic card driver or
+    Operating System
+  * OpenGL Utilities (GLU) ; usually provided by the graphic card driver or
+    Operating System
+  * OpenAL - It's recommended to use the [OpenAL Soft] implementation
+  * [Boost] - 1.56 or later
```

Comment:
Probably 1.59

Reply:
The SDK contains 1.55, CMake required 1.56 and I'm not aware of any feature that we use and isn't provided by 1.56.

Reply:
Mh, seems like the whole version and documentation juggling made me dizzy.  The SDK contains 1.59, the Ubuntu CI uses 1.58.

Reply:
I will raise the minimum requirements in a separate PR to boost 1.58.